packages/core/src/Engine.ts

```diff
@@ -140,9 +150,15 @@ export class Engine extends EventDispatcher {
   private _isDeviceLost: boolean = false;
   private _waitingGC: boolean = false;
 
-  private _animate = () => {
+  private _animate = (...param) => {
     if (this._vSyncCount) {
-      this._requestId = requestAnimationFrame(this._animate);
+      const { _customAnimationFrameRequester: customAnimationFrameRequester } = this;
```

Comment:
You have built the XR concept into the engine. If you enable the XR mode, you can directly use the XR mode update without the need for `_customAnimationFrameRequester`
Maybe you should design XR updtae interface in `XRDevice` or `XRModule`

Reply:
Done.